clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -658,6 +658,10 @@ public void handle(JoinGroupResponse joinResponse, RequestFuture<ByteBuffer> fut
                     AbstractCoordinator.this.generation = new Generation(OffsetCommitRequest.DEFAULT_GENERATION_ID, memberId, null);
                 }
                 future.raise(error);
+            } else if (error == Errors.REBALANCE_IN_PROGRESS) {
+                log.info("JoinGroup failed with a REBALANCE_IN_PROGRESS error, which could indicate a replication timeout " +
```

Comment:
Could we mention in the log that the error is non-fatal, similar to MEMBER_ID_REQUIRED?

Reply:
@abbccdda , thanks for your comment. Just updated. Thanks.